corehq/util/tests/test_quickcache.py

```diff
@@ -283,3 +283,10 @@ def by_name(name, skip_cache=False):
             @skippable_quickcache(['name'], skip_arg='missing')
             def by_name(name):
                 return 'VALUE'
+
+    def test_dict_arg(self):
```

Comment:
Sounds good. I'm not totally sure i know what you mean about testing mismatch, but I changed the test to test whether it was successfully caching on the dict arg, whereas this original test was just testing that passing a dict arg didn't raise an error.